### PR TITLE
AEROGEAR-2630 - Add create client build form.

### PR DESCRIPTION
**Pull request by mikenairn**
* Create jenkins pipeline build config for android applications.
* Enable "Create Build" option only when mobile CI/CD provisioned and client type is Android

**Verification:**
- [ ] Create project with Mobile CI/CD provisioned and an android client
- [ ] Select "Create Build" from the android clients action menu
- [ ] Create a new build for a public repo, debug build type, start the pipeline and verify it completes
- [ ] Create a new build for a private repo, debug build type, start the pipeline and verify it completes
- [ ] Create a new build for a public repo, release build type, start the pipeline and verify it completes
- [ ] Create a new build for a private repo, release build type, start the pipeline and verify it completes

To create a test keystore certificate, use the following:

```
keytool -genkey -v -keystore testcert.keystore -alias testcert -keyalg RSA -keysize 2048 -validity 10000
keytool -importkeystore -srckeystore testcert.keystore -destkeystore testcert.p12 -deststoretype PKCS12 -srcalias testcert
```

Test application with a working Jenkinsfile can be found in this repo https://github.com/mikenairn/helloworld-android-gradle/tree/5xjenkinsfile

### CODE REVIEW

File: app/scripts/controllers/mobileClients.js [hunk 1]
```diff
@@ -73,6 +74,16 @@ angular.module('openshiftConsole')
                 }
                 ctrl.mobileClient = mobileClient;
               }));
+
+              watches.push(DataService.watch(serviceInstanceVersion, context, function (serviceInstances) {
+                $scope.serviceInstances = serviceInstances.by("metadata.name");
+                _.each($scope.serviceInstances, function (serviceInstance) {
```
Comment (sedroche) [inline app/scripts/controllers/mobileClients.js]: Can do:

```.js
ctrl.mobileCIEnabled = _.some($scope.serviceInstances, function(serviceInstance) {
  return /aerogear-digger/.test(serviceInstance.metadata.name);
});
```

**Review by sedroche — APPROVED** [review]

File: app/scripts/controllers/createClientBuild.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+              uri: clientConfig.gitUri,
+              ref: clientConfig.gitRef
+            }
+          },
+          strategy: {
+            jenkinsPipelineStrategy: {
+              jenkinsfilePath: clientConfig.jenkinsfilePath,
+              env: [
+                {
+                  name: 'FH_CONFIG_CONTENT',
```
Comment (ghost) [inline app/scripts/controllers/createClientBuild.js]: Do we still need 'FH_CONFIG_CONTENT'?

File: app/scripts/controllers/mobileClients.js [hunk 3]
```diff
@@ -73,6 +74,16 @@ angular.module('openshiftConsole')
...
+                  if (/aerogear-digger/.test(serviceInstance.metadata.name)) {
+                    ctrl.mobileCIEnabled = true;
```
Comment (ghost) [inline app/scripts/controllers/mobileClients.js]: does not `break` needed here?

**Review by vchepeli — CHANGES_REQUESTED** [review]

File: app/scripts/controllers/mobileClients.js (hunk 1 above)
Comment (mikenairn) [reply app/scripts/controllers/mobileClients.js]: :+1:  Updated

File: app/scripts/controllers/createClientBuild.js (hunk 2 above)
Comment (mikenairn) [reply app/scripts/controllers/createClientBuild.js]: Removed.

If we have no intention of using it in 5.x, i don't think we should have it here. It does however mean the default Jenkinsfile no longer works, so we should update the example to make this optional https://github.com/aerogear/aerogear-digger/blob/master/jenkinsfiles/android_jenkinsfile.groovy#L10.

@odra @maleck13 Does that makes sense to you?

Comment (mikenairn) [reply app/scripts/controllers/createClientBuild.js]: Made changes to this Jenkinsfile to allow it work with the build config generated here https://github.com/mikenairn/helloworld-android-gradle/commit/447f346d35e37d37852c279f4d620a660e40ae88

Comment (maleck13) [reply app/scripts/controllers/createClientBuild.js]: Yes this makes sense to me. Did we create a follow on issue to allow env vars to be specified when creating the build?

**Comment by mikenairn** [comment]
Changing the base branch on here didn't work quite how i expected (it's still listing more changes than there really are), so closing this in favour of https://github.com/aerogear/origin-web-console/pull/43

**Pull request closed without merging**